test/resources/case20_delete_objects/case20_createpod_finalizer.yaml

```diff
@@ -0,0 +1,25 @@
+apiVersion: policy.open-cluster-management.io/v1
+kind: ConfigurationPolicy
+metadata:
+  name: policy-pod-create-withfinalizer
+spec:
+  remediationAction: enforce
+  namespaceSelector:
+    exclude: ["kube-*"]
+    include: ["default"]
+  pruneObjectBehavior: DeleteIfCreated
+  object-templates:
+    - complianceType: musthave
+      objectDefinition:
+        apiVersion: v1
+        kind: Pod
+        metadata:
+          name: nginx-pod-cannot-delete
+          finalizers:
+          - policy.open-cluster-management.io/delete-object
```

Comment:
Nit: a name that makes it clear this is for testing could be better. Otherwise if I read this somewhere I might think it's some new undocumented feature thing.